flink-runtime/src/main/java/org/apache/flink/runtime/state/AbstractKeyedStateBackend.java

```diff
@@ -289,8 +297,11 @@ public KeyGroupRange getKeyGroupRange() {
                 stateDescriptor.initializeSerializerUnlessSet(executionConfig);
             }
             kvState =
-                    TtlStateFactory.createStateAndWrapWithTtlIfEnabled(
-                            namespaceSerializer, stateDescriptor, this, ttlTimeProvider);
+                    LatencyTrackingStateFactory.trackLatencyIfEnabled(
+                            TtlStateFactory.createStateAndWrapWithTtlIfEnabled(
+                                    namespaceSerializer, stateDescriptor, this, ttlTimeProvider),
+                            stateDescriptor,
+                            latencyTrackingStateConfig);
```

Comment:
This will only work for `AbstractKeyedStateBackend` descendants, right? 
Why not to use delegation to enable tracking for any backend?

Reply:
As we already have a `ChangeLogStateBackend` to delegate original keyed state backend, I think it might be a bit too much encapsulated to delegate again.
However, I think your idea also inspires me as we could also let operator state could be tracked if we introduce another `DelegatingStateBackend` here.